server/protocols/jmap-rfc-8621/src/test/scala/org/apache/james/jmap/model/SessionTest.scala

```diff
@@ -135,6 +140,21 @@ class SessionTest extends WordSpec with Matchers {
       } should have message "requirement failed: capabilities should contain urn:ietf:params:jmap:mail capability"
     }
 
+    "throw when missing duplicate capability identifier" in {
```

Comment:
missing or duplicate ?
On another topic i still think that you should not throw.
The verification should be done in the Json validation.
And your constraints be enforced by the type system.
If i understand correctly you must have a core capability and a mail capability, why don't you have a `coreCapability: CoreCapability` and a `mailCapability: MailCapability` in your session object ?